firstvoices/backend/search/indices/dictionary_entry_document.py

```diff
@@ -47,11 +49,20 @@ class Index:
 
 # Signal to update the entry in index
 @receiver(post_save, sender=DictionaryEntry)
-def update_dictionary_entry_index(sender, instance, **kwargs):
+def request_update_dictionary_entry_index(sender, instance, **kwargs):
+    if DictionaryEntry.objects.filter(id=instance.id).exists():
+        update_dictionary_entry_index.apply_async(
+            (instance.id,), countdown=2, link_error=link_error_handler.s()
+        )
+
+
+@shared_task
```

Comment:
If the async tasks are moved to their own file it creates a circular dependency. To avoid this, the tasks are below their corresponding requests.

Reply:
Just an idea for the search folder structuring with the latest changes and to resolve the circular dependency,
Search/
Search/Documents/ -> collection for all the documents
Search/tasks/ -> collection for all the celery tasks 
Search/signals/ -> collection for all the signals

e.g. _for dictionary_entry, there can be a file called dictionary_entry_(document/task/signals) which resides in the designated folder._

Not required to be done now, but just wanted to get team's opinion.